### PR TITLE
OT275-98-List: "GET /categories"

### PR DESCRIPTION
**Pull request by Sieglind**
# [OT275-98](https://alkemy-labs.atlassian.net/browse/OT275-98)

Added Integration tests for "GET/categories"
### HOW HAS THIS BEEN TESTED?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration.

- [ ] Postman.
- [ ] Unit test.
- [x] Integration test.
- [ ] No testing required (only applies for tech task).

**Test Configuration**:

No additional configuration was needed to run these tests

**Screenshots**:

* ....

### CHECKLIST

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have added the new resource in the Postman Collection file.

### CODE REVIEW

File: src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    String categoryName = getRandomCategory().getName();
+
+    mockMvc.perform(get(URL)
+            .header(HttpHeaders.AUTHORIZATION, getAuthorizationTokenForAdminUser()))
+        .andExpect(jsonPath("$.categories", hasSize(2)))
+        .andExpect(jsonPath("$.categories[0].name", equalTo("News")))
+        .andExpect(jsonPath("$.categories[1].name", equalTo(categoryName)))
+        .andExpect(status().isOk());
+
+  }
```
Comment (mkain0) [inline src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java:38]: No hay diferencia entre el test de la linea 19 y este.

File: src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Test
+  public void shouldReturnForbiddenWhenUserHasStandardRole() throws Exception {
+    Long randomCategoryId = getRandomCategoryId();
+
+    mockMvc.perform(get(URL, String.valueOf(randomCategoryId))
```
Comment (mkain0) [inline src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java:44]: UNRELATED: no se requiere un categoryId para listar.

File: src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.hamcrest.Matchers.hasItem;
+
+import com.alkemy.ong.bigtest.BigTest;
+import org.junit.Test;
+import org.springframework.http.HttpHeaders;
+
+public class ListCategoryIntegrationTest extends BigTest {
```
Comment (mkain0) [inline src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java:14]: MISSING CASE: qué pasa si no hay categories en la BD?

**Review by mkain0 — CHANGES_REQUESTED** [review]

File: src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java (hunk 3 above)
Comment (Sieglind) [reply src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java:14]: Asumí erróneamente que la categoría news siempre existía

File: src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  private static final String URL = "/categories";
+
+  @Test
+  public void shouldListCategoriesWithNewsDetailsWhenUserHasAdminRole() throws Exception {
+
+    mockMvc.perform(get(URL)
+            .header(HttpHeaders.AUTHORIZATION, getAuthorizationTokenForAdminUser()))
+        .andExpect(jsonPath("$.page", equalTo(0)))
+        .andExpect(jsonPath("$.size", equalTo(10)))
+        .andExpect(jsonPath("$.totalPages", equalTo(1)))
```
Comment (mkain0) [inline src/test/java/com/alkemy/ong/bigtest/category/ListCategoryIntegrationTest.java:26]: Este endpoint tiene paginación, por lo tanto, también se debe realizar el assert sobre los headers de la response.

**Review by mkain0 — CHANGES_REQUESTED** [review]

**Review by mkain0 — APPROVED** [review]
@Sieglind La forma correcta de testear el header de la response es que muestre valores, si es empty, el test no agrega ningún valor, pues si alguien modifica el helper/util creado para armar los links de la pagina anterior o siguiente, estos test nunca van a fallar y no hay nada que cubra esa lógica de la API.

**Pull request merged**